src/Common/src/System/Net/Logging/SocketsEventSource.cs

```diff
@@ -0,0 +1,268 @@
+            }
+        }
+
+        [NonEvent]
+        internal static void Connected(Socket socket, object localEp, object remoteEp)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            s_log.Connected(LoggingHash.GetObjectName(localEp), LoggingHash.GetObjectName(remoteEp), LoggingHash.HashInt(socket));
+        }
+
+        [Event(CONNECTED_ID, Keywords = Keywords.Default,
+            Level = EventLevel.Informational)]
+        internal unsafe void Connected(string localEp, string remoteEp, int socketHash)
+        {
+            fixed (char* arg1Ptr = localEp, arg2Ptr = remoteEp)
+            {
+                const int SIZEDATA = 3;
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+                dataDesc[0].DataPointer = (IntPtr)(arg1Ptr);
+                dataDesc[0].Size = (localEp.Length + 1) * sizeof(char);
+                dataDesc[1].DataPointer = (IntPtr)(arg2Ptr);
+                dataDesc[1].Size = (remoteEp.Length + 1) * sizeof(char);
+                dataDesc[2].DataPointer = (IntPtr)(&socketHash);
+                dataDesc[2].Size = sizeof(int);
+
+                WriteEventCore(CONNECTED_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [Event(CONNECTED_ASYNC_DNS_ID, Keywords = Keywords.Default,
+            Level = EventLevel.Informational)]
+        internal void ConnectedAsyncDns(int socketHash)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            WriteEvent(CONNECTED_ASYNC_DNS_ID, socketHash);
+        }
+
+        [Event(NOT_LOGGED_FILE_ID, Keywords = Keywords.Default,
+            Level = EventLevel.Informational)]
+        internal unsafe void NotLoggedFile(string filePath, int socketHash, SocketAsyncOperation completedOperation)
+        {
+            fixed (char* arg1Ptr = filePath)
+            {
+                const int SIZEDATA = 3;
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+                dataDesc[0].DataPointer = (IntPtr)(arg1Ptr);
+                dataDesc[0].Size = (filePath.Length + 1) * sizeof(char);
+                dataDesc[1].DataPointer = (IntPtr)(&completedOperation);
+                dataDesc[1].Size = sizeof(int);
+                dataDesc[2].DataPointer = (IntPtr)(&socketHash);
+                dataDesc[2].Size = sizeof(int);
+
+                WriteEventCore(NOT_LOGGED_FILE_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [NonEvent]
+        internal static void Dump(object method, IntPtr bufferPtr, int length)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            if (length > DefaultMaxDumpSize)
+            {
+                length = DefaultMaxDumpSize;
+            }
+            byte[] buffer = new byte[length];
+            Marshal.Copy(bufferPtr, buffer, 0, length);
```

Comment:
Why does this need to copy the data but the other events are only using pointers to existing objects?